packages/auth/src/AuthProvider/AuthProvider.tsx

```diff
@@ -131,11 +134,17 @@ export function createAuthProvider<
     useEffect(() => {
       async function doRestoreState() {
         await authImplementation.restoreAuthState?.()
-        reauthenticate()
+
+        // @MARK(SSR-Auth): Conditionally call reauth, because initial state should come from server (on SSR)
+        // If the inital state didn't come from the server - or was restored already -
+        // reauthenticate will make an call to receive the current user from the server
+        if (!serverAuthState) {
```

Comment:
```suggestion
        // @MARK(SSR-Auth): Conditionally call reauth, because initial state
        // should come from server (on SSR).
        // If the initial state didn't come from the server - or was restored already -
        // reauthenticate will make a call to receive the current user from the server
```